lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp

```diff
@@ -122,48 +120,76 @@ static int start_authentication_information_procedure(
   auth_info_proc->ue_id = ue_id;
   auth_info_proc->resync = auth_info_proc->request_sent;
 
-  bool is_initial_req = !(auth_info_proc->request_sent);
-  auth_info_proc->request_sent = true;
+  rc = amf_authentication_request_sent(ue_id);
+  if (rc != RETURNok) {
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
+
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
+}
+
+int amf_authentication_request_sent(amf_ue_ngap_id_t ue_id) {
+  ue_m5gmm_context_s* ue_mm_context = nullptr;
+  amf_context_t* amf_context = nullptr;
+  char imsi_str[IMSI_BCD_DIGITS_MAX + 1];
+  int rc = RETURNerror;
+  ue_mm_context = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+  uint8_t snni[40] = {0};
+  if (!ue_mm_context) {
+    OAILOG_WARNING(LOG_NAS_AMF,
+                   "AMF-PROC - Failed authentication request the UE due to NULL"
+                   "ue_context\n");
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
+  amf_context = &ue_mm_context->amf_context;
+  nas5g_amf_auth_proc_t* auth_proc =
+      get_nas5g_common_procedure_authentication(amf_context);
 
+  if (!auth_proc) {
+    OAILOG_WARNING(LOG_NAS_AMF, "authentication procedure not present\n");
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
   IMSI64_TO_STRING(amf_context->imsi64, imsi_str, IMSI_LENGTH);
 
   rc = calculate_amf_serving_network_name(amf_context, snni);
   if (rc != RETURNok) {
     OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
   }
 
-  if (is_initial_req) {
+  nas5g_auth_info_proc_t* auth_info_proc =
+      get_nas5g_cn_procedure_auth_info(amf_context);
+  auth_info_proc->request_sent = true;
```

Comment:
Missing NULL check for auth_info_proc before using at 161

Reply:
comment addressed.